noir-projects/noir-contracts/contracts/token_with_refunds_contract/src/types/token_note.nr

```diff
@@ -248,7 +247,12 @@ impl PrivatelyRefundable for TokenNote {
 
         assert_eq(user_point.is_infinite, false);
 
-        // Finally we return the points which represent the note hiding points.
-        (fee_payer_point, user_point)
+        // 4. We no longer need to do any elliptic curve operations with the points so we collapse them to the final
+        // note hashes.
+        let fee_payer_note_hash = fee_payer_point.x;
+        let user_note_hash = user_point.x;
+
+        // 5. Finally we return the hashes.
+        (fee_payer_note_hash, user_note_hash)
```

Comment:
Nice, this very closely mirrors the proposed `finalize` fn from https://github.com/AztecProtocol/aztec-packages/issues/7585